Core/Battleground: made the difference of teams count configurable

FrancescoBorzi opened the pull request:
I just wonder **if** and **how** it should affect [BattlegroundQueue::FillPlayersToBG](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Battlegrounds/BattlegroundQueue.cpp#L496) too.

r00ty-tc committed:
Rooted vehicle movement (#13342 for example)

Corrects issue where certain combinations of actions would cause vehicles (usually cannons/artillary) that should be rooted, to be fully movable

Dire-Tomatoes committed:
Server/Game/AuctionHouse

Fixes world config value of less than 1 being ignored by deposit formula.

Had problems with the last commits and pr, sorry

Update AuctionHouseMgr.cpp

Adds another check to config to handle values over 1 and properly adjust the minimum higher. Thanks jackpoz!

Shauren committed:
Core/Updater: Pipe mysql process output into worldserver log

(cherry picked from commit 2d109b63e2f1d03a961dd58947598f73e4e43ba0)

joschiwald committed:
Core/Updater: fixed host parameter is being ignored

(cherry picked from commit e51b6601779050837fcc87f27579b86c50446fc3)

Commit:
Update authserver.conf.dist

- Added new line for consistency

Shauren committed:
Core/Transports: Fixed accessing deleted instance transports

Thanks @jackpoz for valgrind log and @pete318 for reproduction steps

Closes #15633

(cherry picked from commit 8745e08be020471db886e11a3a7512a751141562)

r4d1sh committed:
Core/Misc: Fix a typo

Killyana committed:
DB/Waypoints: Fix Artificer Daelo path

Updates #15630

Aokromes committed:
DB/Creature: Azuremyst Isle pathing

By Malcrom

Naios committed:
Core/Updater: Fix mysql in path check/search.

Killyana committed:
DB/Event:  Argent Confessor Paletress

http://wow.gamepedia.com/Argent_Confessor_Paletress

Killyana committed:
DB/Creature: Murloc Scout

Closes #15612

FrancescoBorzi committed:
Core/Player: implement ResetCoolDownAfterDuel configurable feature

FrancescoBorzi committed:
Core/Battleground: made the difference of teams count configurable

Shauren commented on src/server/game/Battlegrounds/BattlegroundQueue.cpp:
```diff
@@ -660,8 +660,8 @@ bool BattlegroundQueue::CheckNormalMatch(Battleground* bg_template, Battleground
-        // do not allow to start bg with more than 2 players more on 1 faction
-        if (abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > 2)
+        // do not allow to start bg with more than CONFIG_BATTLEGROUND_INVITATION_DIFF players more on 1 faction
+        if (abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > CONFIG_BATTLEGROUND_INVITATION_DIFF)
```
comparing to a constant (CONFIG_BATTLEGROUND_INVITATION_DIFF) instead of its config value

joschiwald commented on src/server/game/Battlegrounds/BattlegroundQueue.cpp:
```diff
@@ -660,8 +660,8 @@ bool BattlegroundQueue::CheckNormalMatch(Battleground* bg_template, Battleground
-        // do not allow to start bg with more than 2 players more on 1 faction
-        if (abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > 2)
+        // do not allow to start bg with more than CONFIG_BATTLEGROUND_INVITATION_DIFF players more on 1 faction
+        if ((uint32)abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > sWorld-> getIntConfig(CONFIG_BATTLEGROUND_INVITATION_TYPE))
```
`CONFIG_BATTLEGROUND_INVITATION_TYPE` höö???

ShinDarth replied:
sorry typo, fixed

Killyana committed:
DB/Gossip Menu: Add missing gossips for Karazhan's npcs

Move npc_berthold script to the DB
Add missing gossips
Closes #15564
Thanks to @Riztazz and @xjose93

Killyana committed:
DB/SAI: Razormaw

By @pete318
Update #8988

Killyana committed:
DB/Misc: Multiple DB Fixes

Killyana committed:
DB/Creature: Spirit of Olum

Move the script to the DB
Fix conditions for the teleportation
Add the missing teleportation to the Chamber of Command
Fix gossip text

FrancescoBorzi committed:
Core/Ticket keep always trace of GM who resolves the ticket

Runico commented on src/server/game/Battlegrounds/BattlegroundQueue.cpp:
```diff
@@ -660,8 +660,8 @@ bool BattlegroundQueue::CheckNormalMatch(Battleground* bg_template, Battleground
-        // do not allow to start bg with more than 2 players more on 1 faction
-        if (abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > 2)
+        // do not allow to start bg with more than CONFIG_BATTLEGROUND_INVITATION_DIFF players more on 1 faction
+        if ((uint32)abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > sWorld-> getIntConfig(CONFIG_BATTLEGROUND_INVITATION_DIFF))
```
```
sWorld-> getIntConfig(CONFIG_BATTLEGROUND_INVITATION_DIFF)
```
Should be

```
sWorld->getIntConfig(CONFIG_BATTLEGROUND_INVITATION_DIFF)
```

ShinDarth replied:
done, thanks

CrisPj committed:
Core/Scripts: Salanar the Horseman, Into the Realm of Shadows

- Show gossip_menu_option 9739 only if quest 12687 is accepted & not completed
- Cast spell 52693, SPELL_REALM_OF_SHADOWS, when gossip option is selected

Core script by @CreshandKesh (with some modifications)
SQL part by @CreshandKesh and @tkrokli, condition adjustment by @Killyana.

Closes #15603
Closes #15657

joschiwald committed:
Core/Misc: renamed sGossipSelect parameters

Killyana committed:
Scripts/ICC: moved saurfang gossips to db

Closes #14904

Killyana committed:
DB/Creature: Fix Npcflag for High Overlord Saurfang and Muradin Bronzebeard

Killyana committed:
DB/gossip: Add missing gossips to Garrosh Hellscream

Killyana committed:
DB/Creature: Fix  Champion go to bunny movements

Killyana committed:
DB/Creature: Hrothgar Landing Fixups

By @Malcrom

Killyana committed:
DB/Quest: Fix a typo on CompletionText for Dark Storms

By @Espionage724
Closes #15652

Naios committed:
CMake: Fix bad generation of "revision_data.h" not including MYSQL_EXECUTABLE.
* was caused through invoking "FindMySQL" after using ${MYSQL_EXECUTABLE}.
* better re-run CMake if you are affected by the mysql exe wasn't found issue.

Killyana committed:
DB/Creature: Nerub'ar Broodkeeper

Add visual Web Beam on aggro
Fix creature addon for 25 mode
Fix InhabitType issue after landing

Naios committed:
Travis: Add check for update entries not listed in base dumps.
* Will mark the travis build as failed when the database updater is broken.

Naios committed:
DB/Auth: Add missing update 2015_08_21_00_auth.sql to the auth base.

Killyana committed:
DB/Creature: Spire Frostwyrm

Add Spire Frostwyrm spawns and scripts in Icecrown Citadel

DDuarte committed:
Rename SQL after PR merge

Killyana committed:
DB/Creature: Sheddle Glossgleam Event

Shiny Shoes

DDuarte commented on src/server/game/Battlegrounds/BattlegroundQueue.cpp:
```diff
@@ -660,8 +660,8 @@ bool BattlegroundQueue::CheckNormalMatch(Battleground* bg_template, Battleground
-        // do not allow to start bg with more than 2 players more on 1 faction
-        if (abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > 2)
+        // do not allow to start bg with more than CONFIG_BATTLEGROUND_INVITATION_DIFF players more on 1 faction
+        if ((uint32)abs((int32)(m_SelectionPools[TEAM_HORDE].GetPlayerCount() - m_SelectionPools[TEAM_ALLIANCE].GetPlayerCount())) > sWorld->getIntConfig(CONFIG_BATTLEGROUND_INVITATION_DIFF))
```
Do not use C-style casts.
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-casts-named

FrancescoBorzi committed:
Use static cast in BattlegroundQueue.cpp

FrancescoBorzi committed:
Merge remote-tracking branch 'upstream/3.3.5' into bg

Conflicts:
	sql/base/characters_database.sql
